core/convert.rs

```diff
@@ -823,14 +1043,127 @@ where
     scope: &mut v8::PinScope<'s, 'i>,
     value: v8::Local<'s, v8::Value>,
   ) -> Result<Self, Self::Error> {
-    if value.is_undefined() {
+    if value.is_null_or_undefined() {
       Ok(None)
     } else {
       T::from_v8(scope, value).map(|v| Some(v))
     }
   }
 }
 
+impl<'s, T> FromV8Fast<'s> for Option<T>
+where
+  T: FromV8Fast<'s>,
+{
+  fn from_v8(value: v8::Local<'s, v8::Value>) -> Result<Self, Self::Error> {
+    if value.is_null_or_undefined() {
+      Ok(None)
+    } else {
+      <T as FromV8Fast>::from_v8(value).map(|v| Some(v))
+    }
+  }
+}
+
+#[derive(Debug, thiserror::Error, deno_error::JsError)]
+pub enum V8ConvertError {
+  #[class(inherit)]
+  #[error(transparent)]
+  Infallible(#[from] Infallible),
+  #[class(inherit)]
+  #[error(transparent)]
+  DataError(DataError),
+}
+
+impl From<v8::DataError> for V8ConvertError {
+  fn from(value: v8::DataError) -> Self {
+    Self::DataError(value.into())
+  }
+}
+
+impl<'s, T, E> ToV8<'s> for v8::Global<T>
+where
+  Local<'s, T>: TryInto<Local<'s, v8::Value>, Error = E>,
+  E: Into<V8ConvertError>,
+{
+  type Error = V8ConvertError;
+
+  fn to_v8<'i>(
+    self,
+    scope: &mut PinScope<'s, 'i>,
+  ) -> Result<Local<'s, v8::Value>, Self::Error> {
+    let local: Local<'s, T> = Local::new(scope, self);
+    local.try_into().map_err(Into::into)
+  }
+}
+
+impl<'s, T, E> FromV8<'s> for v8::Global<T>
+where
+  Local<'s, v8::Value>: TryInto<Local<'s, T>, Error = E>,
+  E: Into<V8ConvertError>,
+{
+  type Error = V8ConvertError;
+
+  fn from_v8<'i>(
+    scope: &mut PinScope<'s, 'i>,
+    value: Local<'s, v8::Value>,
+  ) -> Result<Self, Self::Error> {
+    Ok(v8::Global::new(
+      scope,
+      value.try_into().map_err(Into::into)?,
+    ))
+  }
+}
+
+impl<'s, T, E> ToV8<'s> for v8::Local<'s, T>
+where
+  Local<'s, T>: TryInto<Local<'s, v8::Value>, Error = E>,
+  E: Into<V8ConvertError>,
+{
+  type Error = V8ConvertError;
+
+  fn to_v8<'i>(
+    self,
+    _scope: &mut PinScope<'s, 'i>,
+  ) -> Result<Local<'s, v8::Value>, Self::Error> {
+    <Self as ToV8Fast>::to_v8(self)
+  }
+}
+
+impl<'s, T, E> ToV8Fast<'s> for v8::Local<'s, T>
+where
+  Local<'s, T>: TryInto<Local<'s, v8::Value>, Error = E>,
+  E: Into<V8ConvertError>,
+{
+  fn to_v8(self) -> Result<Local<'s, v8::Value>, Self::Error> {
+    self.try_into().map_err(Into::into)
+  }
+}
+
```

Comment:
Why is this impl ok? I thought that you can't really return values from fast calls in general, though now I can't actually find the v8 docs on it